### PR TITLE
Execution service by Yurii Kotsiuba

### PR DESCRIPTION
**Pull request by jurok3x**

### CODE REVIEW

File: src/main/java/executor/service/service/impl/ExecutionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public void execute(Scenario scenario, ProxyConfigHolder proxy) {
+        WebDriver webDriver = getWebDriverPrototype(webDriverConfig, proxy);
+        if (webDriver == null) return;
+        scenarioExecutor.execute(scenario, webDriver);
+    }
+
+    private WebDriver getWebDriverPrototype(WebDriverConfig webDriverConfig, ProxyConfigHolder proxyConfigHolder) {
+         WebDriverInitializer webDriverInitializer = new WebDriverInitializerImpl();
+         return webDriverInitializer.getInstance(webDriverConfig, proxyConfigHolder);
+    }
```
Comment (Sokol3783) [inline src/main/java/executor/service/service/impl/ExecutionServiceImpl.java:34]: After merge with ObjectFactory, it's better rewrite call to ObjectFactory

File: src/main/java/executor/service/service/impl/ExecutionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    public ExecutionServiceImpl(ScenarioExecutor scenarioExecutor, WebDriverConfig webDriverConfig) {
+        this.scenarioExecutor = scenarioExecutor;
+        this.webDriverConfig = webDriverConfig;
+    }
+
+    @Override
+    public void execute(Scenario scenario, ProxyConfigHolder proxy) {
+        WebDriver webDriver = getWebDriverPrototype(webDriverConfig, proxy);
+        if (webDriver == null) return;
```
Comment (Sokol3783) [inline src/main/java/executor/service/service/impl/ExecutionServiceImpl.java:27]: it's redunant check

**Review by Sokol3783 — APPROVED** [review]

File: src/main/java/executor/service/service/impl/ExecutionServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import executor.service.model.Scenario;
+import executor.service.model.WebDriverConfig;
+import executor.service.service.ExecutionService;
+import executor.service.service.ScenarioExecutor;
+import executor.service.service.WebDriverInitializer;
+import org.openqa.selenium.WebDriver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExecutionServiceImpl implements ExecutionService {
```
Comment (OlexTul) [inline src/main/java/executor/service/service/impl/ExecutionServiceImpl.java:13]: This class is like mine, see ExecutionService, pull-request [#46](https://github.com/GFLCourses5/worker/pull/46). 
Please, approve ExecutionService, pull-request [#46](https://github.com/GFLCourses5/worker/pull/46).

**Review by OlexTul — CHANGES_REQUESTED** [review]

File: src/main/java/executor/service/service/impl/ExecutionServiceImpl.java (hunk 1 above)
Comment (OlexTul) [reply src/main/java/executor/service/service/impl/ExecutionServiceImpl.java:34]: I don't agree with you, WebDriverInitializer should always be a singleton.
In addition, the infrastructure code has nothing to do in the business logic layer.

**Review by Sokol3783 — APPROVED** [review]

**Review by hromovyi-kostiantyn — APPROVED** [review]

**Pull request closed without merging**